### PR TITLE
feat: enable and disable service providers

### PR DESCRIPTION
**Pull request by xixas**
**Changes**
- This PR will enable users to add only the env values of services that they want to use.
- Minimum of one service needs to be enabled.
- Setup and Usage document updated accordingly.
**Env added**
```sh
ENABLE_SMTP=true
ENABLE_MAILGUN=false
ENABLE_WA360DIALOG=false
```

**Screenshots**
The application exits with an error message if no service is enabled in env.

image
image

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -14,17 +14,20 @@ REDIS_HOST=
 REDIS_PORT=
 
 # SMTP
+ENABLE_MAILGUN=
```
Comment (osm-vishnukyatannawar) [inline .env.example:17]: Why are all same variable?

File: .env.example [hunk 2]
```diff
@@ -14,17 +14,20 @@ REDIS_HOST=
...
 SMTP_HOST=
 SMTP_PORT=
 SMTP_USERNAME=
 SMTP_PASSWORD=
 
 # Mailgun
+ENABLE_MAILGUN=
```
Comment (osm-vishnukyatannawar) [inline .env.example:24]: By same I mean here also it is ENABLE_MAILGUN

File: src/modules/notifications/notifications.controller.ts [hunk 3]
```diff
@@ -20,6 +20,7 @@ export class NotificationsController {
     try {
       const createdNotification =
         await this.notificationService.createNotification(notificationData);
+      this.logger.log('Notification created successfully.');
```
Comment (osm-vishnukyatannawar) [inline src/modules/notifications/notifications.controller.ts:23]: The formatting looks off. Don't we have command to fix it automatically?

**Review by osm-vishnukyatannawar — CHANGES_REQUESTED** [review]

File: .env.example (hunk 1 above)
Comment (xixas) [reply .env.example:17]: Fixed example value.

File: src/modules/notifications/notifications.controller.ts (hunk 3 above)
Comment (xixas) [reply src/modules/notifications/notifications.controller.ts:23]: It's being formatted automatically that way by prettier.

File: .env.example (hunk 2 above)
Comment (xixas) [reply .env.example:24]: Fixed name

**Review by osm-vishnukyatannawar — APPROVED** [review]

**Pull request merged**